agent/eni/iphelperwrapper/iphelper_windows.go

```diff
@@ -0,0 +1,163 @@
+initialNotificationLoop:
+	for {
+		select {
+		case event := <-notificationChannel:
+			if event == InitialNotification {
+				break initialNotificationLoop
+			}
+		case <-time.After(InitialNotificationTimeout):
+			log.Errorf("IPHelper : Initial notification not received within timeout. Closing the Monitor.")
+			if retErr := monitor.Close(); retErr != nil {
+				return errors.Wrapf(retErr, "initial notification not received as well")
+			}
+			return errors.Errorf("initial notification not received while starting monitor")
+		}
+	}
+
+	return nil
+}
+
+// This is the callback function invoked by NotifyIPInterfaceChange Windows API
+// The api returns the information about changed interface as a MIB_IPInterface_Row struct
+// The type of event can be inferred using the notificationType. Since we are concerned about addition of new ENIs,
+// therefore we will pass information to notification channel when the event is of interface addition i.e.1
+func callback(callerContext, row, notificationType uintptr) uintptr {
+
+	// Notification type 1 means addition of a device. InterfaceIndex would always be a positive value
+	if notificationType == 1 {
+		rowdata := *(*MibIPInterfaceRow)(unsafe.Pointer(row))
+		log.Debugf("Interface added with index : %v", rowdata.InterfaceIndex)
+		notificationChannel <- int(rowdata.InterfaceIndex)
+	} else if notificationType == 3 {
+		// Notification type = 3 means Initial notification. This block would be executed only once i.e. when agent starts
+		// Initial notification is sent immediately after registration.
+		// Only after sending this, any other types of notification can be sent
+		// Sending this notification on separate goroutine inorder to avoid deadlock with calling function
+		go func() {
+			time.Sleep(DelayInCallbackMessageForInitialNotification)
+			log.Info("Initial notification received!")
+			// In order to confirm the registration of callback, we will send a negative number on the notification channel
+			notificationChannel <- InitialNotification
+		}()
+	}
+
+	return 0
+}
+
+// This method can be called to cancel the notification callback.
+// Alternatively the notification callback would be cancelled when application exits
+func (monitor *monitoringInterface) Close() (err error) {
+	log.Info("IPHelper : Cancelling the interface change notification callback")
+	if err := monitor.cancelNotifications(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// This method is used to cancel the notification callback for any changes in the interfaces attached to the system
+// If the call is successful then res is 0. Otherwise we will return the corresponding error.
+func (monitor *monitoringInterface) cancelNotifications() error {
+	res, _, _ := funcCancelMibChangeNotify2(uintptr(monitor.notificationHandle))
+
+	if res != 0 {
+		return errors.Errorf("error caused while deregistering from windows notification : error message : %s", syscall.Errno(res))
```

Comment:
Please ensure that all error strings follow a similar pattern across the code base. Avoid using mixed schemes.

Reply:
This error message follows the same scheme as the remaining application. The only addition is "error message:"
Should we remove that part?